modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/util/PowwowUtil.java

```diff
@@ -336,6 +354,85 @@ public static PowwowSubscriptionSender getPowwowSubscriptionSender(
 		return powwowSubscriptionSender;
 	}
 
+	public static String getRecurrenceSummary(Recurrence recurrence, ServiceContext serviceContext) {
+
+		String recurrenceSummary = StringPool.BLANK;
+
+		Locale locale = serviceContext.getLocale();
+
+		StringBuilder sb = new StringBuilder();
+
+		Map<String, Object> valuesMap = new HashMap<>();
+
+		String position = StringPool.BLANK;
+		String weekDay = StringPool.BLANK;
+		String weekDays = StringPool.BLANK;
+
+		if (recurrence.getInterval() == 1) {
+			sb.append(StringUtils.capitalize(recurrence.getFrequency().getValue()));
+		}
+		else {
+			sb.append(LanguageUtil.get(locale, "every") + " ${interval} ${frequencyLabel}");
+
+			valuesMap.put("interval", recurrence.getInterval());
+			valuesMap.put("frequencyLabel", _frequencyMapLabel.get(recurrence.getFrequency()));
```

Comment:
should extract `_frequencyMapLabel.get(recurrence.getFrequency())` to a variable so that it would be easier to understand